### PR TITLE
Fix support for `CombinedLoader` while checking for warning raised with eval dataloaders

### LINKED ISSUE #10866: Warning about shuffle=True despite having shuffle=False in multiple val_dataloader with CombinedLoader
## 🐛 Bug

When working with multiple val_dataloaders with CombinedLoader, the following warning is displayed, despite having `shuffle=False`
```
Warning: 
Your 'val_dataloader' has 'shuffle=True', it is strongly recommended that you turn this off for val/test/predict dataloaders.
```

### To Reproduce
Colab: https://colab.research.google.com/drive/14R9IzJN7xazG1wudz-bnF1LBsmD7vgVt?usp=sharing

### Expected behavior

No warnings should be displayed

### Environment

* CUDA:
	- GPU:
		- Tesla K80
	- available:         True
	- version:           11.1
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.10.0+cu111
	- pytorch-lightning: 1.5.4
	- tqdm:              4.62.3
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.12
	- version:           #1 SMP Sat Jun 5 09:50:34 PDT 2021

cc @justusschock @awaelchli @ninginthecloud

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Fixes #10866

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @justusschock @awaelchli @ninginthecloud

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

File: pytorch_lightning/trainer/data_loading.py [hunk 1]
```diff
@@ -459,3 +458,16 @@ def replace_sampler(dataloader: DataLoader) -> DataLoader:
...
+    @staticmethod
+    def _check_eval_shuffling(dataloader, mode):
+        if (
+            hasattr(dataloader, "sampler")
+            and not isinstance(dataloader.sampler, SequentialSampler)
+            and not isinstance(dataloader.dataset, IterableDataset)
+        ):
+            rank_zero_warn(
+                f"Your `{mode.dataloader_prefix}_dataloader` has `shuffle=True`,"
+                " it is strongly recommended that you turn this off for val/test/predict dataloaders.",
```
Comment (awaelchli) [inline pytorch_lightning/trainer/data_loading.py:471]: Side question, do you remember why? Is that because `drop_last` could be set True? I usually toggle this as a sanity check to verify metrics are the same (shuffling should have no effect).

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/trainer/data_loading.py (hunk 1 above)
Comment (rohitgr7) [reply pytorch_lightning/trainer/data_loading.py:471]: yes, this is one case. another one can be if the user limits the val batches using `limit_val_batches`, so it's recommended to set `shuffle=False` all the time to be on the safe side.

**Review by Borda — APPROVED** [review]
